### ISSUE TITLE
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'constructor')

### ISSUE THREAD

**Issue by AnjeloPeiris711** [cmp-202/ssh2shell#102]
How to fix this error?
![image](https://user-images.githubusercontent.com/51872510/206894062-0450f310-3474-41ff-b224-0f6b09cdcba6.png)

**Comment by cmp-202** [comment]
No idea. 
I have not used promise, I don't have that background understanding of it, 
so I have not tested using it and I don't have a recommended implementation.

As a base level, without promise, there is no error.
How are you implementing promise with ssh2shell?
That JS is generated by the `super hosts` line in the constructor when compiled from the CoffeeScript `/src/ssh2shell.coffee`.

At a guess, have you created an instance `var SSH = new SSH2Shell(host)`? 
If not then `undefined` may result at that point.

Andrew

**Comment by cmp-202** [comment]
Can you try the latest commit and the following code please: [ssh2shell.js](https://github.com/cmp-202/ssh2shell/tree/master/lib/ssh2shell.js)

```javascript
var SSH = new SSH2Shell(host)

var SSHconnect = util.promisify(SSH.connect)
SSHconnect()
  .then(( sessionText ) => {
     console.log ( "-----Callback session text:\n" + sessionText);
     console.log ( "-----Callback end" )})
  .catch((error) => {
     // Handle the error.
     console.log (error);
  });
```

**Comment by cmp-202** [comment]
Sorry I forgot to compile the coffeescript.

**Comment by cmp-202** [comment]
I have published the fix to npm v2.0.8
